[Feat] Component TitleMain 생성

gmlwlsdl opened the pull request:
## #️⃣연관된 이슈

> [Feat] Component TitleMain 생성

## 📝작업 내용

> /src/index.css 생성 후 카카오 폰트를 전역적으로 적용하였습니다.
> 각 페이지에 쓰일 TitMain 컴포넌트를 제작하였습니다.
> 'url' prop에는 해당 div의 background-image 적용해서 사용하시면 됩니다.

### 스크린샷 (선택)
> <img width="484" alt="image" src="https://github.com/user-attachments/assets/0460dd90-6a04-4c3a-b184-a3e8d087dc4f">

## 💬리뷰 요구사항(선택)

> 많은 페이지에 공통적으로 사용되는 컴포넌트라, 빠르게 만들어 올렸습니다! 🙇‍♀️

gmlwlsdl committed:
design: configure index.css

gmlwlsdl committed:
feat: component 'TitMain'

gmlwlsdl committed:
refactor: update css props

gmlwlsdl committed:
design: update font-size of 'EmphTit'

gmlwlsdl committed:
chore: configure Vite and TypeScript for Emotion's css prop support

gmlwlsdl committed:
feat: working on subsidiaryCompany page

gmlwlsdl committed:
feat: working on subsidiaryCompany page

devwontory commented on src/components/ContItem.tsx:
```diff
@@ -0,0 +1,104 @@
+function ContItem({ tit, txt, desc, img, link }: TitItemProps) {
+  return (
+    <>
+      <Cont>
```
단어를 줄여 쓰는 것은 읽는 사람으로 하여금 뜻을 한번 더 생각하게 하기 때문에 최대한 풀네임으로 작성해주세요!

gmlwlsdl replied:
네 알겠습니다!

devwontory commented on src/components/ContItem.tsx:
```diff
@@ -0,0 +1,104 @@
+  return (
+    <>
+      <Cont>
+        <TitItem>{tit}</TitItem>
+        <TxtItem dangerouslySetInnerHTML={{ __html: txt }} />
```
Tit -> TitleItem
Txt 정도는 보편적으로 쓰이기 때문에 괜찮은 것 같습니다.

devwontory commented on src/components/ContItem.tsx:
```diff
@@ -0,0 +1,104 @@
+    <>
+      <Cont>
+        <TitItem>{tit}</TitItem>
+        <TxtItem dangerouslySetInnerHTML={{ __html: txt }} />
+        <DescItem dangerouslySetInnerHTML={{ __html: desc }} />
```
dangerouslySetInnerHTML을 사용하신 이유가 따로 있으신가요?

gmlwlsdl replied:
WrapCommunity.tsx에서 `txt`와 `desc`을 받아올 때, 문장 간 <br /> 처리를 해주어야 하는 부분이 있어 위 방법을 사용하였습니다!

devwontory commented on src/components/ContItem.tsx:
```diff
@@ -0,0 +1,104 @@
+        <TitItem>{tit}</TitItem>
+        <TxtItem dangerouslySetInnerHTML={{ __html: txt }} />
+        <DescItem dangerouslySetInnerHTML={{ __html: desc }} />
+        <WrapLogo href={img} target="_blank" rel="noopener noreferrer">
```
WrapLogo라는 컴포넌트명만 보고는 이 요소가 a 요소인지 알 수 없어요.
이미지는 img 태그로, 로고에 링크가 필요하면 이미지를 a로 감싸주는 것이 더 좋은 것 같아요!

gmlwlsdl replied:
아하 넵 알겠습니다!

devwontory commented on src/components/OutlinkBtn.tsx:
```diff
@@ -0,0 +1,45 @@
+  box-sizing: border-box;
+`
+
+function OutlinkBtn() {
+  return (
+    <>
+      <OutlinkCircle xmlns="http://www.w3.org/2000/svg" viewBox="0 0 44 44">
+        <g css={css({ fill: 'none', fillRule: 'evenodd' })}>
+          <g
+            css={css({
+              transform: 'translate(-640px, -581px) translate(640px, 581px)',
+            })}
+          >
+            <circle
+              cx="22"
+              cy="22"
+              r="22"
+              css={css({ fill: 'rgb(238, 238, 238)' })}
+            />
+            <g>
+              <path d="M0 0H14V14H0z" transform="translate(15 15)" />
+              <path
+                d="M2.792 1.67L12.354 1.67 12.354 11.233M12.354 1.67L2.143 11.881"
+                transform="translate(15 15)"
+                strokeWidth="1.5"
+                css={css({ stroke: 'black' })}
+              />
+            </g>
+          </g>
+        </g>
+      </OutlinkCircle>
+    </>
+  )
+}
```
컴포넌트명은 Btn인데 코드를 보면 버튼의 역할을 수행하고 있지 않아요!

gmlwlsdl replied:
넵 수정하도록 하겠습니다

devwontory commented on src/components/WrapCommunity.tsx:
```diff
@@ -0,0 +1,212 @@
+                link="https://t1.kakaocdn.net/kakaocorp/kakaocorp/admin/community/c6658b5b018b00001.png"
+              />
+            </Li>
+            <Li>
+              <ContItem
+                tit="카카오페이"
+                txt="누구에게나 이로운 금융서비스를 만듭니다."
+                desc="카카오페이는 모두에게 이로운 금융 서비스를 만드는 ‘생활 금융 플랫폼’입니다.  <br /> 기존 금융의 번거로움에서 벗어나 우리 생활 전반에 이로운 흐름을 만들고자 합니다. 
+             <br />결제, 송금부터 대출, 보험, 투자 등 금융까지 서비스 폭을 넓히며 일상의 모든 금융이 <br /> 카카오페이 하나로 되기 위해 노력하고 있습니다."
+                img="https://www.kakaopay.com"
+                link="https://t1.kakaocdn.net/kakaocorp/kakaocorp/admin/community/c6664cfa018b00001.png"
+              />
+            </Li>
+            <Li>
+              <ContItem
+                tit="카카오뱅크"
+                txt="같지만 다른 은행"
+                desc="카카오뱅크는 사람과 은행의 만남이 더 쉽게, 더 자주 일어나야 한다고 믿습니다. <br /> 사용자 중심의 혁신적인 기술을 통해 일상 속 어떤 순간에서도 유용한 나만의 은행. <br /> 카카오뱅크는 또 하나의 은행이 아닌 '은행의 새로운 시작' 입니다."
+                img="https://www.kakaobank.com/"
+                link="https://t1.kakaocdn.net/kakaocorp/kakaocorp/admin/community/c666f9a4018b00001.png"
+              />
+            </Li>
+            <Li>
+              <ContItem
+                tit="카카오모빌리티"
+                txt="우리의 기술로 일상을 움직입니다."
+                desc="카카오모빌리티는 생활 속 이동이 더 편리해지고, 그 방법도 더 다양해져야 한다고 생각합니다. <br /> 이를 위해 택시를 시작으로 대리, 내비, 바이크, 주차, 차량관리 서비스, 퀵/택배 등 우리 일상에  <br /> 필요한 모든 이동 서비스들을 만들어가며, AI와 자율주행 등 미래를 위한 기술을 연구합니다."
+                img="https://www.kakaomobility.com/"
+                link="https://t1.kakaocdn.net/kakaocorp/kakaocorp/admin/community/c6679cc7018b00001.png"
+              />
+            </Li>
+            <Li>
+              <ContItem
+                tit="카카오스타일"
+                txt="나로 가득한 라이프스타일"
+                desc="카카오스타일은 모든 사람이 나만의 특별한 스타일을 가지고 있고, 내가 좋아하는 무언가를  <br /> 발견했을 때의 즐거움이 나의 일상을 더욱 나답게 만든다고 믿습니다.  <br /> 내 마음을 읽은 듯한 개인화 추천으로 누구나 나만의 스타일을 쉽게 찾도록 도움으로써 다양한 <br /> 스타일이 공존하는 세상을 만들겠습니다."
+                img="https://kakaostyle.com/"
+                link="https://t1.kakaocdn.net/kakaocorp/kakaocorp/admin/community/c66852ad018b00001.png"
+              />
+            </Li>
```
반복적인 패턴은 데이터를 따로 배열로 빼고 map으로 렌더링할 수 있어요.
```tsx
const contents = [{ title: 카카오, text: '사람을 이해하는 기술로, ...', description: '카카오는 사람에 대한 이해를 바탕으로 ...', image: 'https://careers...', link: 'https://t1.kakaocdn...' }, ...]

...

contents.map(({title, text, description, image, link}) => (
  <li>
    <ContentItem
      key={??}
      title={title}
      text={text}
      description={description}
      image={image}
      link={link} />
  </li>
))
```

gmlwlsdl replied:
넵 적용하도록 하겠습니다!

devwontory replied:
배열로 옮기시면서 desc 자료구조를 고민해보시면 위에서 이야기했던 dangerouslySetInnerHTML도 없앨 수 있을 것 같습니다~

devwontory commented on src/main.tsx:
```diff
@@ -1,10 +1,15 @@
-const router = createBrowserRouter([{ path: '/', element: <Main /> }])
+const router = createBrowserRouter([
+  { path: '/', element: <Main /> },
+  { path: '/about/subsidiaryCompany', element: <SubsidiaryCompany /> },
```
```ts
{
  path: "/page-a",
  element: <PageA/>,
  // children에 중첩라우팅을 배열로 추가합니다.
  children: [
    {
      path: "Children-1",
      element: <Children1/>,
    },
    {
      path: "Children-2",
      element: <Children2/>,
    },
  ],
},
```

gmlwlsdl replied:
중첩라우팅 적용하겠습니다!

devwontory reviewed (changes requested):
컴포넌트명은 길어도 괜찮아요.
뭔가를 줄여쓰는 것은 좋은 패턴은 아니라고 생각합니다!

gmlwlsdl committed:
refactor: code review feedback

- 컴포넌트명 수정 'ContItem' > 'ContentItem'
- 컴포넌트명 수정 'OutlinkBtn' > 'OutlinkSvg'
- 컴포넌트명 수정 'TitMain' > 'TitleMain'
- 직관적 변수명, 태그 적용
- OutlinkSvg hover 추가
- map 적용
- 중첩 라우팅 적용

Refs: #9

gmlwlsdl committed:
chore: renamed SubsidiaryCompany

- index.css 위치 변경

gmlwlsdl committed:
Rename subsidiaryCompany.tsx to SubsidiaryCompany.tsx

gmlwlsdl committed:
chore: Update main.tsx

devwontory reviewed (approved):
LGTM
